tests/deposit_reclaim_test.cairo

```diff
@@ -0,0 +1,285 @@
+
+    usdc.approve(token_bridge.contract_address, 100);
+    token_bridge.deposit(usdc_address, 100, snf::test_address());
+
+    snf::start_cheat_block_timestamp_global(5);
+    token_bridge.deposit_cancel_request(usdc_address, 100, snf::test_address(), 2);
+
+    snf::start_cheat_block_timestamp_global(
+        starknet::get_block_timestamp() + DELAY_TIME.try_into().unwrap() + 10
+    );
+
+    snf::start_cheat_caller_address_global(contract_address_const::<'user2'>());
+    token_bridge.deposit_reclaim(usdc_address, 100, snf::test_address(), 2);
+}
+
+
+#[test]
+fn deposit_with_message_reclaim_ok() {
+    let (token_bridge, mut spy, messaging_mock) = deploy_token_bridge_with_messaging();
+    let usdc_address = deploy_erc20("usdc", "usdc");
+    let usdc = IERC20Dispatcher { contract_address: usdc_address };
+
+    enroll_token_and_settle(token_bridge, messaging_mock, usdc_address);
+
+    let mut calldata = ArrayTrait::new();
+    'param1'.serialize(ref calldata);
+    'param2'.serialize(ref calldata);
+
+    usdc.approve(token_bridge.contract_address, 100);
+    token_bridge.deposit_with_message(usdc_address, 100, snf::test_address(), calldata.span());
+
+    snf::start_cheat_block_timestamp_global(5);
+    token_bridge
+        .deposit_with_message_cancel_request(
+            usdc_address, 100, snf::test_address(), calldata.span(), 2
+        );
+
+    snf::start_cheat_block_timestamp_global(
+        starknet::get_block_timestamp() + DELAY_TIME.try_into().unwrap() + 10
+    );
+
+    let initial_user_balance = usdc.balance_of(snf::test_address());
+    token_bridge
+        .deposit_with_message_reclaim(usdc_address, 100, snf::test_address(), calldata.span(), 2);
+    assert(
+        usdc.balance_of(snf::test_address()) == initial_user_balance + 100, 'deposit not recieved'
+    );
+
+    let expected_deposit_cancel = TokenBridge::DepositWithMessageCancelRequest {
+        sender: snf::test_address(),
+        token: usdc_address,
+        amount: 100,
+        appchain_recipient: snf::test_address(),
+        message: calldata.span(),
+        nonce: 2
+    };
+
+    let expected_deposit_reclaim = TokenBridge::DepositWithMessageReclaimed {
+        sender: snf::test_address(),
+        token: usdc_address,
+        amount: 100,
+        appchain_recipient: snf::test_address(),
+        message: calldata.span(),
+        nonce: 2
+    };
+
+    spy
+        .assert_emitted(
+            @array![
+                (
+                    token_bridge.contract_address,
+                    Event::DepositWithMessageCancelRequest(expected_deposit_cancel)
+                ),
+                (
+                    token_bridge.contract_address,
+                    Event::DepositWithMessageReclaimed(expected_deposit_reclaim)
+                )
+            ]
+        );
+}
+
+#[test]
+#[should_panic(expected: ('CANCELLATION_NOT_ALLOWED_YET',))]
+fn deposit_with_message_reclaim_delay_not_reached() {
```

Comment:
let's also add this to the accept_deposit issue?

Reply:
Like dont `deposit` and `deposit_with_message` these 2 functions (`deposit_reclaim` and `deposit_with_message_reclaim`) do not call a common internal function which can be tested to avoid the replication. Although still have added this to the issue to further think on this how we could improvise the testcases here